### PR TITLE
Document how to read data from a stream layer

### PR DESCRIPTION
**Pull request by HalynaDumych**
Add documentation that describes how to subscribe to a stream layer,
get data from it, and unsubscribe.

Relates-To: OLPEDGE-1726

Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: docs/dataservice-read-from-stream-layer-example.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+2. In the **build** folder, build the example project.
+
+   ```bash
+   cmake --build . --target dataservice-example
+   ```
+
+3. Execute the example project.
+
+   ```bash
+   ./examples/dataservice-example --example read --key_id "here.access.key.id" --key_secret "here.access.key.secret" --catalog "catalog"
```
Comment (andescu) [inline docs/dataservice-read-from-stream-layer-example.md]:
```suggestion
   ./examples/dataservice-example --example read_stream --key_id "here.access.key.id" --key_secret "here.access.key.secret" --catalog "catalog"
```

File: docs/dataservice-read-from-stream-layer-example.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+   ```bash
+   ./examples/dataservice-example --help
+   ```
+
+After building and running the example project, you see the following information:
+
+- `edge-example-catalog` &ndash; a catalog description
+- `versioned-world-layer` &ndash; a layer description
+- `Request partition data - Success, data size - 3375` - a success message that displays the size of data retrieved from the requested partition.
```
Comment (andescu) [inline docs/dataservice-read-from-stream-layer-example.md]: These are not valid for stream layer reading.
Please get some valid logs from @LiubovDidkivska.

File: docs/dataservice-read-from-stream-layer-example.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+```bash
+[INFO] read-example - Catalog description: edge-example-catalog
+[INFO] read-example - Layer 'versioned-world-layer' (versioned): versioned-world-layer
+[INFO] read-example - Layer contains 1 partitions
+[INFO] read-example - Partition: 1
+[INFO] read-example - Request partition data - Success, data size - 3375
+```
```
Comment (andescu) [inline docs/dataservice-read-from-stream-layer-example.md]: Same here.

File: docs/dataservice-read-from-stream-layer-example.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+   ```cpp
+   auto request = olp::dataservice::read::SubscribeRequest()
+                     .WithSubscriptionMode(olp::dataservice::read::SubscribeRequest::SubscriptionMode::kSerial));
+   ```
+
+5. Call the `Subscribe` method with the `SubscribeRequest` parameter.
+
+   ```cpp
+   client::CancellationToken Subscribe(SubscribeRequest request,
```
Comment (andescu) [inline docs/dataservice-read-from-stream-layer-example.md]:
```suggestion
   client::CancellableFuture<SubscribeResponse> Subscribe(
      SubscribeRequest request);
```
As you also use the future based APIs below for `Poll()` and `GetData()`

File: docs/dataservice-read-from-stream-layer-example.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+6. Call the `Poll` method.
+
+   ```cpp
+   client::CancellableFuture<PollResponse> Poll();
+   ```
+
+   You get messages with the layer data and partition metadata. The `Poll` method also commits the offsets, so you can continue polling new messages.
+
+   If the data size is less than 1 MB, the data field is populated. If the data size is greater than 1 MB, you get a data handle that points to the object stored in the blob store.
```
Comment (andescu) [inline docs/dataservice-read-from-stream-layer-example.md]:
```suggestion
   If the data size is less than 1 MB, the `data` field is populated. If the data size is greater than 1 MB, you get a data handle that points to the object stored in the blob store.
```

File: docs/dataservice-read-from-stream-layer-example.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
+# Read from a Stream Layer (Example)
+
+On this page, find instructions on how to build and run the example project on different platforms and get catalog and partition metadata, as well as partition data using the HERE Open Location Platform (OLP) SDK for C++.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply docs/dataservice-read-from-stream-layer-example.md]: Changed

File: docs/dataservice-read-from-stream-layer-example.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+Before you run the example project, authorize to HERE OLP:
+
+1. On the [Apps & keys](https://platform.here.com/admin/apps) page, copy your application access key ID and access key secret.
+   For instructions on how to get the access key ID and access key secret, see the [Get Credentials](https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) section in the Terms and Permissions User Guide.
+
+2. In `examples/main.cpp`, replace the placeholders with your access key ID, access key secret, and Here Resource Name (HRN) of the catalog.
+   **Note:**  You can also specify these values using the command line options.
+
+   ```cpp
+   AccessKey access_key{};  // Your access key ID and access key secret.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply docs/dataservice-read-from-stream-layer-example.md]: Changed

File: docs/dataservice-read-from-stream-layer-example.md (hunk 1 above)
Comment (HalynaDumych) [reply docs/dataservice-read-from-stream-layer-example.md]: Changed

File: docs/dataservice-read-from-stream-layer-example.md (hunk 2 above)
Comment (HalynaDumych) [reply docs/dataservice-read-from-stream-layer-example.md]: Discussed and changed.

File: docs/dataservice-read-from-stream-layer-example.md (hunk 3 above)
Comment (HalynaDumych) [reply docs/dataservice-read-from-stream-layer-example.md]: Discussed and changed.

File: docs/dataservice-read-from-stream-layer-example.md (hunk 4 above)
Comment (HalynaDumych) [reply docs/dataservice-read-from-stream-layer-example.md]: Changed

File: docs/dataservice-read-from-stream-layer-example.md (hunk 5 above)
Comment (HalynaDumych) [reply docs/dataservice-read-from-stream-layer-example.md]: Changed

**Pull request merged**